### ISSUE TITLE
Course Generator Question

### ISSUE THREAD

**Issue by patch110** [Courseplay/Courseplay_FS22#1033]
Hi not sure if an issue or not as game doesn't actually crash but course generator never seems to finish when i choose by-pass islands either simple or circle. when i  say never works i have waited in excess of 5 mins for generator to finish. without by-pass islands selected it works just fine but field does have islands so really do need the islands option. The map in question is Calm Land on Giants Modhub
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/8098981/log.txt)
.

**Comment by Tensuko** [comment]
Since there is no Error and it seems like only unpatient I can't see this as a bug or something.
Screenshots of the Field and Position of your starting point ?

**Comment by patch110** [comment]
![fsScreen_2022_02_18_17_29_46](https://user-images.githubusercontent.com/32200890/154733095-2227deb7-f231-4a5e-8484-de407ade170d.png)
![fsScreen_2022_02_18_17_31_28](https://user-images.githubusercontent.com/32200890/154733213-d66800cd-a1f7-416a-a259-b22ec84bf3e1.png)

**Comment by pvaiko** [comment]
I'm not familiar with this map, but the problem here seems to be that it does not have built in fields, everything you want to make a field you have to plow, if I'm not mistaken. So this is a custom field you created, that is, this isn't really a field (other than the brown area at the perimeter what you seem to have plowed). 

The course generator can only detect islands if they are surrounded by a field (something which you can work on, other than with a plow) already.

So this use case just won't work. I guess the only way around this would be if we implemented something to allow drawing islands within a custom field.

**Comment by pvaiko** [comment]
In this particular case you could work around it by creating two custom fields, so that island is between the two fields, generate a plow course for both and once it is done, merge the two fields.

**Comment by Tensuko** [comment]
Oh okay I also thought there is a real field behind it.
Okay nevermind then it can't work.

So do it as @pvaiko said and split it into too to create it and then combine them.
